File: remote/src/main/java/com/kelsos/mbrc/mappers/ArtistMapper.java
package com.kelsos.mbrc.mappers;

import android.support.annotation.NonNull;
import com.kelsos.mbrc.dao.ArtistDao;
import com.kelsos.mbrc.dao.views.GenreArtistView;
import com.kelsos.mbrc.domain.Artist;
import com.kelsos.mbrc.dto.library.ArtistDto;
import java.util.List;
import rx.Observable;

public class ArtistMapper {
  public static List<ArtistDao> map(List<ArtistDto> data) {
    return Observable.from(data).map(ArtistMapper::map).toList().toBlocking().first();
  }

  public static ArtistDao map(ArtistDto dto) {
    ArtistDao dao = new ArtistDao();
    dao.setId(dto.getId());
    dao.setName(dto.getName());
    dao.setDateAdded(dto.getDateAdded());
    dao.setDateUpdated(dto.getDateUpdated());
    dao.setDateDeleted(dto.getDateDeleted());
    return dao;
  }

  @NonNull public static Artist map(ArtistDao dao) {
    return new Artist(dao.getId(), dao.getName());
  }

  public static List<Artist> mapData(List<ArtistDao> data) {
    return Observable.from(data).map(ArtistMapper::map).toList().toBlocking().first();
  }

  public static List<Artist> mapGenreArtists(List<GenreArtistView> genreArtistViews) {
    return Observable.from(genreArtistViews).map(ArtistMapper::mapGenreArtist).toList().toBlocking().first();
  }

  @NonNull public static Artist mapGenreArtist(GenreArtistView genreArtist) {
   return new Artist(genreArtist.getId(), genreArtist.getName());
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/mappers/ArtistMapper.kt
package com.kelsos.mbrc.mappers

import com.kelsos.mbrc.data.dao.ArtistDao
import com.kelsos.mbrc.data.views.GenreArtistView
import com.kelsos.mbrc.domain.Artist
import com.kelsos.mbrc.dto.library.ArtistDto

object ArtistMapper {
  fun map(data: List<ArtistDto>): List<ArtistDao> {
    return data.map { map(it) }
  }

  fun map(dto: ArtistDto): ArtistDao {
    val dao = ArtistDao()
    dao.id = dto.id
    dao.name = dto.name
    dao.dateAdded = dto.dateAdded
    dao.dateUpdated = dto.dateUpdated
    dao.dateDeleted = dto.dateDeleted
    return dao
  }

  fun map(dao: ArtistDao): Artist {
    return Artist(dao.id, dao.name)
  }

  fun mapData(data: List<ArtistDao>): List<Artist> {
    return data.map { map(it) }
  }

  fun mapGenreArtists(genreArtistViews: List<GenreArtistView>): List<Artist> {
    return genreArtistViews.map { mapGenreArtist(it) }
  }

  fun mapGenreArtist(genreArtist: GenreArtistView): Artist {
    return Artist(genreArtist.id, genreArtist.name!!)
  }
}
